From 5d95f6d76c046c36bb8599c8c56d1818faeb4369 Mon Sep 17 00:00:00 2001
From: Thor Thayer <thor.thayer@linux.intel.com>
Date: Fri, 2 Feb 2018 11:44:13 -0600
Subject: [PATCH 028/151] FogBugz #532583-3: mtd: a10sr: Add A10SR HWMON

commit  8721044a05fa3cbb3d68fe4206f6983313e8e360 from
https://github.com/altera-opensource/linux-socfpga.git
branch is socfpga-5.4.64-lts

Add HWMON to the A10 System Resource Multi-Function Device.
This fixes a previous patch
['commit b5678c1b894d ("FogBugz #411698-2: Add Altera A10-SR
                        power supply alarms")']
that removed this change when cherry-picked.

Signed-off-by: Thor Thayer <thor.thayer@linux.intel.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
Integrated-by: Jun Zhang <jun.zhang@windriver.com>
---
 drivers/mfd/altera-a10sr.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/mfd/altera-a10sr.c b/drivers/mfd/altera-a10sr.c
index a3bf64f9afd1..5430f273f8fb 100644
--- a/drivers/mfd/altera-a10sr.c
+++ b/drivers/mfd/altera-a10sr.c
@@ -18,6 +18,10 @@
 #include <linux/spi/spi.h>
 
 static const struct mfd_cell altr_a10sr_subdev_info[] = {
+	{
+		.name = "altr_a10sr_hwmon",
+		.of_compatible = "altr,a10sr-hwmon",
+	},
 	{
 		.name = "altr_a10sr_gpio",
 		.of_compatible = "altr,a10sr-gpio",
-- 
2.26.1

